File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] arr) {
        
    		if (arr == null || arr.length == 0)
			return Integer.MIN_VALUE;

		int count = 1, maj = arr[0];

		for (int i = 0; i < arr.length; i++) {
			if (arr[i] == maj) {
				count++;
			} else {
				count--;
			}

			if (count == 0) {
				maj = arr[i];
				count = 1;
			}
		}

		return maj;
    }
}

File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    if len(nums) == 0 {
        return 0;
    }    
    
    count := 0
    maj := nums[0]
    
    for i := 0; i < len(nums); i++ {
        if nums[i] == maj {
            count++
        }else{
            count--
        }
        
        if count == 0 {
            maj = nums[i]
            count = 1
        }
    }
    
    return maj
}